src/api/app/controllers/webui/package_controller.rb

```diff
@@ -549,28 +549,7 @@ def branch_dialog
   end
 
   def branch
-    authorize @package, :branch?
-    # FIXME: This authorize isn't in sync with the permission checks of BranchPackage. And the created
-    #        project might differ from the one we check here.
-    authorize Project.new(name: User.current.branch_project_name(@project)), :create?
-
-    branched_package = BranchPackage.new(project: @project.name, package: @package.name).branch
-    created_project_name = branched_package[:data][:targetproject]
-    created_package_name = branched_package[:data][:targetpackage]
-
-    Event::BranchCommand.create(project: @project.name, package: @package.name,
-                                targetproject: created_project_name, targetpackage: created_package_name,
-                                user: User.current.login)
-    redirect_to(package_show_path(project: created_project_name, package: created_package_name),
-                notice: "Successfully branched package")
-  rescue BranchPackage::DoubleBranchPackageError
-      redirect_to(package_show_path(project: User.current.branch_project_name(@project), package: @package),
-                  notice: 'You have already branched this package')
-  rescue APIException => e
-      redirect_back(fallback_location: root_path, error: e.message)
-  end
-
-  def save_new_link
+    authorize Project.new(name: params[:target_project]), :create?
```

Comment:
Why? Wouldn't BranchPackage anyway throw an DoubleBranch exception?
And we didn't do it before either :cry: